### PR TITLE
add `marshal` serialization tests for `float`, `double`, `decimal`

### PR DESCRIPTION
**Pull request by illia-li**

### CODE REVIEW

File: marshal_10_decimal_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/gocql/gocql/marshal/tests/serialization"
+)
+
+func TestMarshalDecimal(t *testing.T) {
+	marshal := func(i interface{}) ([]byte, error) { return Marshal(NativeType{proto: 4, typ: TypeDecimal}, i) }
+	unmarshal := func(bytes []byte, i interface{}) error {
+		return Unmarshal(NativeType{proto: 4, typ: TypeDecimal}, bytes, i)
+	}
+
+	brokenTypes := utils.GetTypes(inf.Dec{}, (*inf.Dec)(nil))
```
Comment (dkropachev) [inline marshal_10_decimal_test.go:19]: Can I ask you to add a comment that breafly describe why these types are broken

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: marshal_10_decimal_test.go (hunk 1 above)
Comment (illia-li) [reply marshal_10_decimal_test.go]: Done

File: marshal_10_decimal_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+func TestMarshalDecimal(t *testing.T) {
+	marshal := func(i interface{}) ([]byte, error) { return Marshal(NativeType{proto: 4, typ: TypeDecimal}, i) }
+	unmarshal := func(bytes []byte, i interface{}) error {
+		return Unmarshal(NativeType{proto: 4, typ: TypeDecimal}, bytes, i)
+	}
+
+	// In cases with data len <4 unmarshal returns an error: "inf.Dec needs at least 4 bytes, while value has only 0"
+	// So, unmarshalls 'nil' and 'zero' 'decimal' currently impossible.
```
Comment (dkropachev) [inline marshal_10_decimal_test.go:20]:
```suggestion
	// Unmarshal does not support deserialization of 'nil' and 'zero' 'into `inf.Dec`.
```

Comment (dkropachev) [reply marshal_10_decimal_test.go:20]: @illia-li , i don't see technical problem to make it happen, can I ask you either create an issue to fix that or outline why it is better not to do it

Comment (illia-li) [reply marshal_10_decimal_test.go]: Done in this version:
// Unmarshal does not support deserialization of `decimal` with `nil` and `zero` `value len` 'into `inf.Dec`.

Comment (dkropachev) [reply marshal_10_decimal_test.go]: What about issue ? are you going to create it ?

File: marshal_10_decimal_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package gocql
```
Comment (dkropachev) [inline marshal_10_decimal_test.go:1]:
```suggestion
package gocql_test
```

**Review by dkropachev — CHANGES_REQUESTED** [review]
One more change and let's merge it

File: marshal_10_decimal_test.go (hunk 2 above)
Comment (illia-li) [reply marshal_10_decimal_test.go]: in progress

Comment (illia-li) [reply marshal_10_decimal_test.go]: Issue done #282

File: marshal_10_decimal_test.go (hunk 3 above)
Comment (illia-li) [reply marshal_10_decimal_test.go]: Done

**Review by dkropachev — APPROVED** [review]

**Pull request merged**